src/transformers/image_processing_utils_fast.py

```diff
@@ -236,6 +232,134 @@ class DefaultFastImageProcessorPreprocessKwargs(DefaultFastImageProcessorInitKwa
+            default_name = default.strip("`").split("self.")[-1]
+            if hasattr(obj, default_name):
+                header_replaced = header.replace(default, f"`{repr(getattr(obj, default_name))}`")
+            else:
+                header_replaced = header
+
+            kwargs_dict[current_kwarg] = {
+                "header_replaced": header_replaced,
+                "header": header,  # also keep the original header for the preprocess method
+                "desc": [],
+            }
+        elif line.strip() != "":
+            raise ValueError(f"Unexpected line in docstring: {line}. Check if the indentation is correct.")
+
+        i += 1
+    return kwargs_dict
+
+
+def customize_docstrings(
+    init_header: str, custom_docstring: str = "", custom_preprocess_docstring: Optional[str] = None
+):
+    """
+    Decorator to customize the docstrings of the `__init__` and `preprocess` methods of a class.
+
+    Args:
+        init_header (`str`):
+            The header to use for the `__init__` method docstring.
+        custom_docstring (`str`, *optional*, defaults to `""`):
+            The custom docstring to use for the `__init__` method. One can provide only the docs for the arguments that
+            need to be customized. The rest of the docstring is generated from the default docstring.
+        custom_preprocess_docstring (`str`, *optional*):
+            The custom docstring to use for the `preprocess` method. If not provided, the `custom_docstring` is used
+            for the `preprocess` method as well.
+    """
+
+    def docstring_decorator(obj):
+        # Parse the base docstring, replace the defaults with the actual values from the object (for the init method)
+        base_init_kwargs = get_kwargs_dict_from_docstring(BASE_IMAGE_PROCESSOR_FAST_DOCSTRING, obj)
+        base_preprocess_kwargs = get_kwargs_dict_from_docstring(BASE_IMAGE_PROCESSOR_FAST_DOCSTRING_PREPROCESS, obj)
+
+        # Parse the given custom docstring and replace the defaults with the actual values from the object (for the init method)
+        custom_kwargs = get_kwargs_dict_from_docstring(custom_docstring, obj)
+        # If a custom_preprocess_docstring is not provided, use the given custom_docstring for the preprocess method as well
+        if custom_preprocess_docstring:
+            custom_preprocess_kwargs = get_kwargs_dict_from_docstring(custom_preprocess_docstring, obj)
+        else:
+            custom_preprocess_kwargs = custom_kwargs
+
+        # Remove the doc for unused kwargs
+        unused_kwargs = getattr(obj, "unused_kwargs", [])
+        for kwarg in unused_kwargs:
+            base_init_kwargs.pop(kwarg, None)
+            base_preprocess_kwargs.pop(kwarg, None)
+
+        # Merge the base and custom kwargs, putting the custom ones first
+        all_init_kwargs = {}
+        for kwarg, values in custom_kwargs.items():
+            if kwarg not in base_init_kwargs:
+                all_init_kwargs[kwarg] = values
```

Comment:
Ok, sounds good, usually, it should be the same order as in the signature, but we don't have an explicit signature here.